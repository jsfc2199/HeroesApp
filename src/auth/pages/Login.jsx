import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import { AuthContext } from "../context/AuthContext";

export const Login = () => {
  const {login} = useContext(AuthContext) //del contexto que da el provider obtenemos el estado del login
  const navigate = useNavigate();

  const onLogin = () => {

    login('Juan Franco') //hacemos el dispatch del login

    navigate("/",{
      replace:true
    });
  };

  return (
    <div className="container mt-5">
      <h1>LoginPage</h1>
      <hr />
      <button className="bnt btn-primary" onClick={onLogin}>
        Login
      </button>
    </div>
  );
};
