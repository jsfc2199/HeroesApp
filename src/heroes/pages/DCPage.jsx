import HeroList from "../components/HeroList"


export const DCPage = () => {
  return (
    <>
      <h1>Dc Comics</h1>
      <hr />
      <HeroList publisher = 'DC Comics'/>
    </>
  )
}


