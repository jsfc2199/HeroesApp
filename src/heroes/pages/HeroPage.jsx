import { Navigate, useNavigate, useParams } from "react-router-dom";
import { getHeroesById } from "../helpers";
import { useMemo } from "react";

export const HeroPage = () => {
  const { id } = useParams();

  const hero = useMemo(() => getHeroesById(id), [id]); //la dependencia es el id

  const navigate = useNavigate();

  const onNavigateBack = () => {
    navigate(-1); //-1 devuelve a la pagina inmediatamente anterior
  }

  if (!hero) {
    return <Navigate to="/marvel" />;
  }

  return (
    <div className="row mt-5">
      <div className="col-4">
        <img
          src={`/assets/heroes/${id}.jpg`}
          alt={hero.superhero}
          className="img-thumbnail animate__animated animate__fadeInLeft"
        />
      </div>
      <div className="col-8">
        <h4>{hero.superhero}</h4>
        <ul className="list-group list-group-flush">
          <li className="list-group-item">
            <b>Alter ego:</b> {hero.alter_ego}
          </li>
          <li className="list-group-item">
            <b>Alter ego:</b> {hero.publisher}
          </li>
          <li className="list-group-item">
            <b>Alter ego:</b> {hero.first_appearance}
          </li>
        </ul>

        <h5 className="mt 3"> Characters</h5>
        <p>{hero.characters}</p>

        <button className="btn btn-outline-primary" onClick={onNavigateBack}>
          Regresar
        </button>
      </div>
    </div>
  );
};
