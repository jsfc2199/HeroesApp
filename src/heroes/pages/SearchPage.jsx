import { HeroCard } from './../components/HeroCard';
export const SearchPage = () => {

  return (
    <>
      <h1>Search</h1>
      <hr />

      <div className="row">
      <div className="col-4">
        <h4>Searching</h4>
        <hr />
        <form action="">
          <input
            type="text"
            placeholder="search a hero"
            name="searchText"
            autoComplete="off"
          />
        </form>

        <button className="btn btn-outline-primary mt-1">Search</button>
      </div>

      <div className="col-7">

        <h4>Results</h4>
        <hr />
        <div className="alert alert-primary">
          Search a Hero
        </div>

        <div className="alert alert-danger">
          No hero with <b>ABC</b>
        </div>

        <HeroCard />

      </div>
      </div>
    </>
  );
};
