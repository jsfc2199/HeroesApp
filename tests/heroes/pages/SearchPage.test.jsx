/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { SearchPage } from './../../../src/heroes/pages/SearchPage';

const mockedUseNavigat = jest.fn()

jest.mock('react-router-dom' , () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigat
}));

describe('Pruebas en el search page', () => {

    beforeEach(() => jest.clearAllMocks());

  test('debe de mostrarse correctamente con valores por defectos', () => {
    const {container} = render(
        <MemoryRouter>
            <SearchPage/>
        </MemoryRouter>
    )

    expect(container).toMatchSnapshot()
  })

  test('debe de mostrarse a bataman y el input con el valor del query string', () => {
    render(
        <MemoryRouter initialEntries={['/search?q=batman']}>
            <SearchPage/>
        </MemoryRouter>
    )

    const input = screen.getByRole('textbox')
    expect(input.value).toBe('batman')

    
    const img = screen.getByRole('img')
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg')

    const alert = screen.getByLabelText('alert-danger')
    expect(alert.style.display).toBe('none')
  })

  test('debe de mostrar un error si no se encuentra el heroe', () => {
    render(
        <MemoryRouter initialEntries={['/search?q=batman123']}>
            <SearchPage/>
        </MemoryRouter>
    )

    screen.debug()
    const alert = screen.getByLabelText('alert-danger')
    expect(alert.style.display).toBe('')
  })

  test('debe de llamar el navigate a la pantalla nueva ', () => {
    render(
        <MemoryRouter initialEntries={['/search?q=batman123']}>
            <SearchPage/>
        </MemoryRouter>
    )

    //llenamos el input de informacion
    const input = screen.getByRole('textbox')
    fireEvent.change(input, {target: {name: 'searchText', value: 'superman'}})

    //hacemos el submit del formulario que nos debe de llevar a la otra pagina
    const form = screen.getByRole('form')
    fireEvent.submit(form)

    expect(mockedUseNavigat).toHaveBeenCalledWith('?q=superman')
  })
  
  
  
})
