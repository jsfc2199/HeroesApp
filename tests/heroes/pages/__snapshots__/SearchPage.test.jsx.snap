// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el search page debe de mostrarse correctamente con valores por defectos 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-4"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          name="searchText"
          placeholder="search a hero"
          type="text"
          value=""
        />
      </form>
      <button
        class="btn btn-outline-primary mt-1"
      >
        Search
      </button>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a Hero
      </div>
      <div
        aria-label="alert-danger"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
